src/35search.js

```diff
@@ -8,1133 +8,1103 @@
-					if (r.length > 0) {
-						if (sidx + 1 + 1 > selectors.length) {
-							return stack;
+	*/
+			var val, // temp values use many places
+				nest, // temp value used many places
+				r, // temp value used many places
+				sel = selectors[sidx];
+			//		console.log(sel);
+			//		if(!alasql.srch[sel.srchid]) {
+			//			throw new Error('Selector "'+sel.srchid+'" not found');
+			//		};
+
+			var INFINITE_LOOP_BREAK = alasql.options.loopbreak || 100000;
+
+			if (sel.selid) {
+				// TODO Process Selector
+				if (sel.selid === 'PATH') {
+					var queue = [{node: value, stack: []}];
+					var visited = {};
+					//var path = [];
+					var objects = alasql.databases[alasql.useid].objects;
+					while (queue.length > 0) {
+						var q = queue.shift();
+						var node = q.node;
+						var stack = q.stack;
+						var r = processSelector(sel.args, 0, node);
+						if (r.length > 0) {
+							if (sidx + 1 + 1 > selectors.length) {
+								return stack;
+							} else {
+								var rv = [];
+								if (stack && stack.length > 0) {
+									stack.forEach(function (stv) {
+										rv = rv.concat(processSelector(selectors, sidx + 1, stv));
+									});
+								}
+								return rv;
+								//							return processSelector(selectors,sidx+1,stack);
+							}
 						} else {
-							var rv = [];
-							if (stack && stack.length > 0) {
-								stack.forEach(function (stv) {
-									rv = rv.concat(processSelector(selectors, sidx + 1, stv));
-								});
+							if (typeof visited[node.$id] !== 'undefined') {
+								continue;
+							} else {
+								//							console.log(node.$id, node.$out);
+								visited[node.$id] = true;
+								if (node.$out && node.$out.length > 0) {
+									node.$out.forEach(function (edgeid) {
+										var edge = objects[edgeid];
+										var stack2 = stack.concat(edge);
+										stack2.push(objects[edge.$out[0]]);
+										queue.push({
+											node: objects[edge.$out[0]],
+											stack: stack2,
+										});
+									});
+								}
 							}
-							return rv;
-							//							return processSelector(selectors,sidx+1,stack);
 						}
+					}
+					// Else return fail
+					return [];
+				}
+				if (sel.selid === 'NOT') {
+					var nest = processSelector(sel.args, 0, value);
+					//console.log(1,nest);
+					if (nest.length > 0) {
+						return [];
 					} else {
-						if (typeof visited[node.$id] !== 'undefined') {
-							continue;
+						if (sidx + 1 + 1 > selectors.length) {
+							return [value];
 						} else {
-							//							console.log(node.$id, node.$out);
-							visited[node.$id] = true;
-							if (node.$out && node.$out.length > 0) {
-								node.$out.forEach(function (edgeid) {
-									var edge = objects[edgeid];
-									var stack2 = stack.concat(edge);
-									stack2.push(objects[edge.$out[0]]);
-									queue.push({
-										node: objects[edge.$out[0]],
-										stack: stack2,
-									});
+							return processSelector(selectors, sidx + 1, value);
```

Comment:
Looking so much forward to us massaging the code more to get out of if-else nested hell